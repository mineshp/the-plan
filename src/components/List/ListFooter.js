import React from 'react';
import PropTypes from 'prop-types';
import { Table, Button, Icon } from 'semantic-ui-react';

const ListFooter = ({ numColumns, handleAddItem, handleSubmit }) => (
    <Table.Footer fullWidth>
        <Table.Row>
            <Table.HeaderCell colSpan={numColumns}>
                <Button floated="left" color="blue" icon labelPosition="left" size="small" onClick={handleAddItem}>
                    <Icon name="plus" /> Add Item
                </Button>
                <Button
                    floated="right"
                    icon
                    labelPosition="left"
                    color="green"
                    size="small"
                    onClick={handleSubmit}
                >
                    <Icon name="save" /> Save
                </Button>
            </Table.HeaderCell>
        </Table.Row>
    </Table.Footer>
);

ListFooter.propTypes = {
    numColumns: PropTypes.number.isRequired,
    handleAddItem: PropTypes.func.isRequired,
    handleSubmit: PropTypes.func.isRequired
};

export default ListFooter;
