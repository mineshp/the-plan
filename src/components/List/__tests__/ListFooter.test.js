import React from 'react';
import renderer from 'react-test-renderer';
import ListFooter from '../ListFooter';

const mockHandleAddItem = jest.fn();
const mockHandleSubmit = jest.fn();
const mockNumColumns = 3;

describe('ListHeadings', () => {
    it('renders footer correctly', () => {
        const tree = renderer.create(<ListFooter
            numColumns={mockNumColumns}
            handleAddItem={mockHandleAddItem}
            handleSubmit={mockHandleSubmit}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
