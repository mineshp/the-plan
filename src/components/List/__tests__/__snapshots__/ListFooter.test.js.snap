// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeadings renders footer correctly 1`] = `
<tfoot
  className="full-width"
>
  <tr
    className=""
  >
    <th
      className=""
      colSpan={3}
    >
      <button
        className="ui blue small icon left floated left labeled button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        <i
          aria-hidden="true"
          className="plus icon"
        />
         Add Item
      </button>
      <button
        className="ui green small icon right floated left labeled button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        <i
          aria-hidden="true"
          className="save icon"
        />
         Save
      </button>
    </th>
  </tr>
</tfoot>
`;
